lib/csv.rb

```diff
@@ -995,8 +1024,13 @@ def instance(data = $stdout, **options)
             options.values_at(*DEFAULT_OPTIONS.keys.sort_by { |sym| sym.to_s })
 
       # fetch or create the instance for this signature
-      @@instances ||= Hash.new
-      instance = (@@instances[sig] ||= new(data, **options))
+      instances_hash = if defined?(Ractor)
+                         Ractor.current[:__CSV_instances_hash__] ||= Hash.new
+                       else
+                         @@instances ||= Hash.new
+                       end
+
+      instance = (instances_hash[sig] ||= new(data, **options))
```

Comment:
Ractor users should not use `CSV.instance`.
They should create your own CSV instance.

Reply:
I believe this change basically makes CSV.instance find a Ractor-local instance. Could you explain more about why this might be a problem? Thanks!

Reply:
Yes. `CSV.instance` can work in Ractor with this change.
It's just API design decision that we don't recommend (support) to use `CSV.instance` in Ractor.
We can just add "This API isn't Ractor-safe" to document here.

Reply:
Okay, I understand!

Reply:
Could you document it in this pull request?